plugins/modules/snap.py

```diff
@@ -0,0 +1,338 @@
+rc:
+    description: The return code.
+    returned: if the command failed.
+    type: int
+stdout:
+    description: The standard output.
+    returned: if the command failed.
+    type: str
+stderr:
+    description: The standard error.
+    returned: if the command failed.
+    type: str
+
+'''
+cmd = ['snap']  # declare a global variable for snap
+
+expectPrompts = {
+    "reset": "/usr/bin/expect -c \"spawn snap -r; \
+            expect \\\"Do you want me to remove these directories \\(y|n\\): \\\"; \
+            send \\\"y\\r\\\"; \
+            expect \\\"Nothing to clean up\\\";\""
+}
+
+
+def build_snap_command(module):
+    """
+    Called the build snap command and check memory greated than 8MB
+    arguments:
+     module (AnsibleModule): The Ansible module instance.
+    return:
+        CMD - command which is succesfully cretaed
+
+    """
+    if check_memory_availbilty():
+        if module.params['general_option_selection']:
+            snap_general_option(module)
+    else:
+        print("this is very low memory less than 8 mb")
+    return cmd
+
+
+def snap_general_option(module):
+
+    """
+    Build the snap command with specify option
+    arguments:
+     module (AnsibleModule): The Ansible module instance.
+    return:
+        True - when command succesfully cretaed
+    """
+    global cmd
+    if module.params['reset']:
+        cmd = ['snap']
+        cmd += ['-r']
+        return True
+    else:
+        if module.params['all_info']:
+            cmd = ['snap']
+            cmd += ['-a']
+            return True
+        elif module.params['hacmp']:
+            cmd = ['snap']
+            cmd += ['-e']
+            return True
+        else:
+            if module.params['file_system_info']:
+                cmd += ['-f']
+            if module.params['live_kernel']:
+                cmd += ['-U']
+            if module.params['installation_info']:
+                cmd += ['-i']
+            if module.params['kernel_info']:
+                cmd += ['-k']
+            if module.params['workload_manager_info']:
+                cmd += ['-w']
+            if module.params['general_info']:
+                if module.params['security_info']:
+                    cmd += ['-S']
+                cmd += ['-g']
+            elif module.params['hardware_info']:
+                cmd += ['-H']
+            if module.params['collects_dump']:
+                cmd += ['-D']
+            if module.params['compress']:
+                cmd += ['-c']
+    return True
+
+
+def check_memory_availbilty():
+    """
+    Check the memroy is greater than 8 MB
+    arguments:
+      None
+    return:
+        True - If the memory greater than 8 MB
+        False - If the memory does not greater than 8 MB.
+    """
+
+    stat = shutil.disk_usage("/")
+    free_space_mb = stat.free // (1024 * 1024)
+    if free_space_mb < 8:
+        raise RuntimeError("Insufficient memory: At least 8MB free space is required to run the '-a' option.")
```

Comment:
Would it be a better idea to use fail_json instead of raising the exception through python ?